Automattic/wp-calypso
# Earn: Refactor plans modal to TypeScript

## Proposed Changes

* Refactors the Earn > RecurringPaymentsPlanAddEditModal component to TypeScript. This includes moving away from the old connect() method.
* This is partially a follow up to a prior PR. See PR and [comment here](https://github.com/Automattic/wp-calypso/pull/80296#discussion_r1293866314).

## Testing Instructions



Functionality and appearance should be the same before and after this PR. We're just testing that there are no regressive breakages after the refactor.

1) Test Launchpad
   - Start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro, choose the Paid option, and proceed to Launchpad. You can also use an existing paid newsletter site that still has full screen launchpad if you have one.
   - If needed, connect Stripe. If you want to avoid fully connecting Stripe, add `define( 'USE_STORE_SANDBOX', true ); `to wp-content/mu-plugins/0-sandbox.php in your sandbox, and sandbox public-api.wordpress.com and the domain of your test site. Then click the link to go to Stripe and you'll see a button to skip the setup. When done, you should return to Launchpad. 
   - Click the Launchpad task to `Create paid newsletter`. Confirm the Plans Modal opens and looks like it is supposed to look. Create a plan, save, and confirm you see a success notification in the upper right corner of your screen. Navigate to `http://calypso.localhost:3000/earn/payments-plans/YOURDOMAIN` and confirm the plan you just created is visible on your plans page.

2) Test the Earn page
   - You'll need a Stripe-connected test site. You can use the one created just above.
   - Go to `http://calypso.localhost:3000/earn/payments-plans/YOURDOMAIN`
   - Click the `Add a new payment plan` button. Confirm the plans modal opens and looks how it is supposed to look. 
   - Add a plan, save, and confirm the plan is added after the modal closes. 
   - Click to edit a plan, and try changing various options and just be sure they all save as expected. 
   - Click to delete a plan, and confirm that works.


## Pre-merge Checklist
